packages/app/components/Queue/QuestionForm.tsx

```diff
@@ -0,0 +1,143 @@
+  name: string;
+  questionType: QuestionType;
+  waitTime: number;
+  status: string;
+}
+
+export default function QuestionForm() {
+  const [questionType, setQuestionType] = useState<QuestionType | undefined>(
+    undefined
+  );
+  const [questionText, setQuestionText] = useState<string | undefined>(
+    undefined
+  );
+
+  // on question type change, update the question type state
+  const onCategoryChange = (e: RadioChangeEvent) => {
+    setQuestionType(e.target.value);
+  };
+
+  // on question text change, update the question text state
+  const onQuestionTextChange = (
+    event: React.ChangeEvent<HTMLTextAreaElement>
+  ) => {
+    setQuestionText(event.target.value);
+  };
+
+  // on button submit click, conditionally choose to go back to the queue
+  const onClickSubmit = () => {
+    if (!!questionType && questionText && questionText !== "") {
+      // todo: submit question (send http server whatnot)
+      // bring the user back to the queue page, assuming it was successful
+      Router.push("/queue");
+    }
+  };
+
+  // on button leave queue click, delete question draft
+  const onClickLeave = () => {
+    // TODO: delete question (send http request)
+  };
+
+  // compute the button-type of the submit button, based on if everything is filled out
+  const computeQuestionType = () => {
+    if (!!questionType && questionText && questionText !== "") {
+      return "primary";
+    } else {
+      return "disabled";
+    }
+  };
+
+  return (
+    <Container>
+      <Alert
+        message="You are currently 12th in queue"
+        description="Your spot in queue has been temporarily reserved. Please describe your question to finish joining the queue."
+        type="success"
+        showIcon
+      />
+
+      <Title>Describe your question</Title>
+
+      <QuestionText>What category does your question fall under?</QuestionText>
+      <Radio.Group
+        onChange={onCategoryChange}
+        buttonStyle="solid"
+        style={{ marginBottom: 48 }}
+      >
+        <Radio.Button value={QuestionType.Concept}>Concept</Radio.Button>
+        <Radio.Button value={QuestionType.Testing}>Testing</Radio.Button>
+        <Radio.Button value={QuestionType.Bug}>Bug</Radio.Button>
+        <Radio.Button value={QuestionType.Setup}>Setup</Radio.Button>
+        <Radio.Button value={QuestionType.Other}>Other</Radio.Button>
+      </Radio.Group>
+
+      <QuestionText>What do you need help with?</QuestionText>
+      <Input.TextArea
+        placeholder="I’m having trouble understanding list abstractions, particularly in the
+        context of Assignment 5."
+        autoSize={{ minRows: 3 }}
+        onChange={onQuestionTextChange}
+      />
+      <QuestionCaption>
+        Be as descriptive and specific as possible in your answer. If your
+        question matches another student’s, your wait time may be reduced.
+      </QuestionCaption>
+
+      <div>
+        <FormButton type={computeQuestionType()} onClick={onClickSubmit}>
```

Comment:
Everything looks great. The only change I see that could be made is on this line. 
```suggestion
        <FormButton type="primary" disabled={!questionType || !questionText || questionText === ""} onClick={onClickSubmit}>
```
It might be slightly more readable from a code standpoint and will show the button as disabled in addition to automatically changing the color. You could then remove the `computeQuestionType()` function which took me a second to figure out what it was doing at first.

Reply:
Good catch, I didn't know there was a disabled property on button.